### PR TITLE
(#121) Bump Cake references to v3.0.0

### LINKED ISSUE #121: Update Cake.Coveralls add-in to target Cake v3.0.0
[Cake 3.0.0](https://cakebuild.net/blog/2022/11/cake-v3.0.0-released) has been released  🎉

### PR DESCRIPTION
**Pull request by nils-a**
## Description
Bumped Cake references to v3.0.0

## Related Issue

fixes #121 

## Motivation and Context

#121

## How Has This Been Tested?

manually 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: Source/Cake.Coveralls.Tests/Cake.Coveralls.Tests.csproj [hunk 1]
```diff
@@ -1,14 +1,14 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>net5.0;net6.0;netcoreapp3.1</TargetFrameworks>
+    <TargetFrameworks>net6.0;net7.0</TargetFrameworks>
```
Comment (raman-m) [inline Source/Cake.Coveralls.Tests/Cake.Coveralls.Tests.csproj:4]: Was there a specific reason for not including the `net8.0` framework in the release?

Comment (raman-m) [reply Source/Cake.Coveralls.Tests/Cake.Coveralls.Tests.csproj:4]: The understanding is that `net8.0` will be included in the [v4.0.0 release](https://www.nuget.org/packages/Cake.Core/4.0.0#supportedframeworks-body-tab) of the Core, right?

Comment (nils-a) [reply Source/Cake.Coveralls.Tests/Cake.Coveralls.Tests.csproj:4]: > Was there a specific reason for not including the `net8.0` framework in the release?

Yes-and-no. I always try to keep the TMFs for the tests close to the TFMs for the "real" package. `Cake.Coveralls.csproj` also lists  `net6.0;net7.0` as TFMs.

> The understanding is that `net8.0` will be included in the [v4.0.0 release](https://www.nuget.org/packages/Cake.Core/4.0.0#supportedframeworks-body-tab) of the Core, right?

Correct. According to [CakeContrib.Guidelines#TargetFramework](https://cake-contrib.github.io/CakeContrib.Guidelines/guidelines/TargetFramework) addins targeting Cake v4.0.0 should use `net6.0;net7.0;net8.0` as TFMs.

Comment (raman-m) [reply Source/Cake.Coveralls.Tests/Cake.Coveralls.Tests.csproj:4]: Thank you for the explanations! The [net8.0](https://github.com/cake-contrib/Cake.Coveralls/pull/124/files#diff-ca729cd1a5d9ff1740b8b58738ee274c6ccd09a2e73d0ccba1f99f05c5a4c7c2R4) TFM has been added in PR #124.

**Pull request merged**